astropy/coordinates/angles/core.py

```diff
@@ -582,7 +582,11 @@ class Latitude(Angle):
 
     def __new__(cls, angle, unit=None, **kwargs):
         # Forbid creating a Lat from a Long.
-        if isinstance(angle, Longitude):
+        if (
+            isinstance(angle, Longitude)
+            or isinstance(angle, str)
+            and angle.endswith(("E", "W"))
```

Comment:
Human readers would benefit from an additional pair of parenthesis, even if Python doesn't need them (also in `Longitude.__new__()):
```suggestion
            isinstance(angle, Longitude)
            or (isinstance(angle, str) and angle.endswith(("E", "W")))
```